source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -120,7 +141,10 @@ void CacheFilter::onHeaders(LookupResult&& result) {
       return;
     }
     if (result.content_length_ > 0) {
-      remaining_body_.emplace_back(0, result.content_length_);
+      if (remaining_body_.empty()) {
+        // no range was added to remaining_body_, will add entire body to response
+        remaining_body_.emplace_back(0, result.content_length_);
```

Comment:
I don't understand what's happening here. Is space in remaining_body_ being reserved with 0s, for getBody() to then fill in?

Reply:
`getBody()` was already structured to treat ranges.
The original implementation simply adds a range from byte 0 until the end (i.e., `content_length_`), and lets `getBody()` fill the body with this _full range_.

My change here is to first insert the range that was requested specifically; and then only include the full range if there's nothing added yet.

Does this answer your question?

Reply:
I see. I realize the code was already structured like this before this PR, but I don't like range request logic happening here in the generic `Ok` case. Maybe the helper function refactoring I suggested can rearrange that out?

Probably only in a review could someone as unfamiliar as me be paying close enough attention (without going as far as getting familiar) to care about remaining_body_ while also not realizing it's not chunks of data, so it probably doesn't really matter. But (although maybe it's used elsewhere in a way that would make this a bad choice) I'm thinking a better name might be `remaining_ranges_`, both to describe that it's concerned entirely with range logic, and to avoid making it sound like some kind of buffer.